docs/data/sources/google-tag-manager-client.md

```diff
@@ -3,17 +3,256 @@ title: Google Tag Manager Template - Amplitude Analytics Browser SDK
+|`quantity` | Required. `number`. The quantity of products purchased. For version after marketing-analytics-browser-gtm 0.5.4 , we auto assign `revenue = quantity * price`. &&&&&& | `1` |
+|`price` | Required. `number`. The price of the products purchased, and this can be negative. Note: revenue = quantity* price. | `null` |
+|`revenue_type` | Optional, but required for revenue verification. `string`. The revenue type (for example, tax, refund, income). | `null` |
+|`event properties`| Optional. `{ [key: string]: any }`. An object of event properties to include in the revenue event. | `null` |
+
+#### flush
+
+The flush method prompts the client to instantly send buffered events. [More Details](/docs/data/sdks/typescript-browser/#flush-the-event-buffer). There's no need to manually call the `flush `tag type, it will automatically be triggered based on either `flushIntervalMillis` or `flushQueueSize`, whichever comes first. To avoid event loss due to browser closure, consider enabling the `sendBeacon` transport in configuration step or set the transport type to `sendBeacon` when `pagehide`. [More details](../../data/sdks/typescript-browser/migration/#patterns).  Alternatively, you could reduce the `flushQueueSize` and `flushIntervalMillis` according to your event traffic load, ensuring events don't get stuck on the client side.
+
+#### setUserId
+
+If you want to reset the userId and deviceId after logout, please check `reset` tag type instead.
+
+| Name | Description | Default Value |
+| --- | --- | --- |
+| `User ID`| `undefined` or `string`. Set the userId. Leave empty if you want to set the userId to undefined. | `undefined` |
+
+#### setDeviceId
+
+If you want to reset the userId and deviceId after logout, please check `reset` tag type instead.
+
+| Name | Description | Default Value |
+| --- | --- | --- |
+| `Device ID`| `string`. Set the deviceId for the current user. Amplitude will assign an unique identifier for the deviceId by default. For the cross domain use case, Amplitue will auto capture the `deviceId` from URL parameter and assign the value to deviceId. Otherwise, this is not recommended unless you know what you are doing. [More Details](https://www.docs.developers.amplitude.com/data/sdks/typescript-browser/#custom-device-id) | `UUID` |
+
+#### setSessionId
+
+Session logic is auto handled by Amplitude. You might need to set the sessionId to `-1` if you want to out of session control. 
+Please make sure the value is in milliseconds since epoch (Unix Timestamp) or `-1`.
+
+[More details](../../data/sdks/typescript-browser/#custom-session-id)
+
+#### reset 
+
+`reset` will be commonly used when a user has been logged out. It includes 2 operations which are `setUserId(undefined)` and `setDeviceId(UUID())`.
+
+[More details](/sdks/typescript-browser/#reset-when-user-logs-out).
+
+#### setOptOut
+
+Check the `Opt current user out of tracking` checkbox to opt user out of tracking. 
```

Comment:
How about adding some cookie behavior description here or at init tag type

Cookies are created on initialization. So even if a user opts out, cookies are still created. If you want to don't want to track cookies, for example, before user consents, you should wait to init

Reply:
I added this in init section.